invenio_rdm_records/requests/community_submission.py

```diff
@@ -69,13 +69,16 @@ class DeclineAction(actions.DeclineAction):
 
     def execute(self, identity, uow):
         """Execute action."""
-        # Unset review from record (still accessible from request)
-        # This means that the receiver (curator) won't have access anymore
+        # Keeps the record and the review connected so the user can see the
+        # outcome of the request
+        # The receiver (curator) won't have access anymore to the draft
         # The creator (uploader) should still have access to the record/draft
         draft = self.request.topic.resolve()
-        draft.parent.review = None
         super().execute(identity, uow)
 
+        draft.parent.review = self.request
```

Comment:
Minor: It ought to already be assigned, so it seems strange this line is neeeded. You'll still need the line below (``RecordCommitOp``), but can you move it before the ``super().execute()``